platform/android/src/native_map_view.cpp

```diff
@@ -70,117 +71,50 @@ NativeMapView::NativeMapView(jni::JNIEnv& _env,
         return;
     }
 
-    auto& fileSource = mbgl::android::FileSource::getDefaultFileSource(_env, jFileSource);
-
-    // Create a renderer
-    auto renderer = std::make_unique<Renderer>(*this, pixelRatio, fileSource, *threadPool,
-                                               GLContextMode::Unique,
-                                               jni::Make<std::string>(_env, _programCacheDir));
+    mbgl::FileSource& fileSource = mbgl::android::FileSource::getDefaultFileSource(_env, jFileSource);
 
     // Create a renderer frontend
-    rendererFrontend = std::make_unique<AndroidRendererFrontend>(std::move(renderer),
-                                                                 *this,
-                                                                 [this] { this->invalidate(); });
+    rendererFrontend = std::make_unique<AndroidRendererFrontend>(pixelRatio, fileSource,
+                                                                 *threadPool,
+                                                                 jni::Make<std::string>(_env,
+                                                                                        _programCacheDir),
+                                                                 [this] { this->requestRender(); },
```

Comment:
Remove this indirection: arrange to pass a `jni::Object<GLSurfaceView>` directly, and have `AndroidRendererFrontend` call `requestRender` and `queueEvent` on it.